testsupport/wait/member.go

```diff
@@ -246,6 +246,117 @@ func (a *MemberAwaitility) WaitForUserAccount(t *testing.T, name string, criteri
 	return userAccount, err
 }
 
+// SpaceRequestWaitCriterion a struct to compare with a given SpaceRequest
+type SpaceRequestWaitCriterion struct {
+	Match func(request *toolchainv1alpha1.SpaceRequest) bool
+	Diff  func(request *toolchainv1alpha1.SpaceRequest) string
+}
+
+// WaitForSpaceRequest waits until there is a SpaceRequest available with the given name, namespace, spec and the set of status conditions
+func (a *MemberAwaitility) WaitForSpaceRequest(t *testing.T, namespacedName types.NamespacedName, criteria ...SpaceRequestWaitCriterion) (*toolchainv1alpha1.SpaceRequest, error) {
+	var spaceRequest *toolchainv1alpha1.SpaceRequest
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		obj := &toolchainv1alpha1.SpaceRequest{}
+		if err := a.Client.Get(context.TODO(), namespacedName, obj); err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		spaceRequest = obj
+		return matchSpaceRequestWaitCriterion(obj, criteria...), nil
+	})
+	// no match found, print the diffs
+	if err != nil {
+		a.printSpaceRequestWaitCriterionDiffs(t, spaceRequest, criteria...)
+	}
+	return spaceRequest, err
+}
+
+// UntilSpaceRequestHasTierName returns a `SpaceRequestWaitCriterion` which checks that the given
+// SpaceRequest has a specific tierName
+func UntilSpaceRequestHasTierName(expected string) SpaceRequestWaitCriterion {
+	return SpaceRequestWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.SpaceRequest) bool {
+			return actual.Spec.TierName == expected
+		},
+		Diff: func(actual *toolchainv1alpha1.SpaceRequest) string {
+			return fmt.Sprintf("expected tier name to be '%s'\nbut it was '%s'", expected, actual.Spec.TierName)
+		},
+	}
+}
+
+// UntilSpaceRequestHasTargetClusterRoles returns a `SpaceRequestWaitCriterion` which checks that the given
+// SpaceRequest has the expected cluster roles
+func UntilSpaceRequestHasTargetClusterRoles(expected []string) SpaceRequestWaitCriterion {
+	return SpaceRequestWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.SpaceRequest) bool {
+			return reflect.DeepEqual(expected, actual.Spec.TargetClusterRoles)
+		},
+		Diff: func(actual *toolchainv1alpha1.SpaceRequest) string {
+			return fmt.Sprintf("expected space roles to match:\n%s", Diff(expected, actual.Spec.TargetClusterRoles))
+		},
+	}
+}
+
+// UntilSpaceRequestHasStatusTargetClusterURL returns a `SpaceRequestWaitCriterion` which checks that the given
+// SpaceRequest has the expected .Status.TargetClusterURL value
+func UntilSpaceRequestHasStatusTargetClusterURL(expected string) SpaceRequestWaitCriterion {
+	return SpaceRequestWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.SpaceRequest) bool {
+			return expected == actual.Status.TargetClusterURL
+		},
+		Diff: func(actual *toolchainv1alpha1.SpaceRequest) string {
+			return fmt.Sprintf("expected space roles to match:\n%s", Diff(expected, actual.Status.TargetClusterURL))
```

Comment:
```suggestion
			return fmt.Sprintf("expected target cluster URL to match:\n%s", Diff(expected, actual.Status.TargetClusterURL))
```

Reply:
Thanks! applied in [32f2a7c](https://github.com/codeready-toolchain/toolchain-e2e/pull/685/commits/32f2a7cc9eb5347d00c19dbbd3a44a54ecbcfb4c)